tests/suites/test_suite_ecdsa.function

```diff
@@ -14,7 +14,7 @@ void ecdsa_prim_random( int id )
     mbedtls_ecp_point Q;
     mbedtls_mpi d, r, s;
     rnd_pseudo_info rnd_info;
-    unsigned char buf[66];
+    unsigned char buf[MBEDTLS_MD_MAX_SIZE];
```

Comment:
This isn't the only place we have a buffer of size 66 in this file. Should we fix the other places as well?

Reply:
This is the only place I found.

Reply:
I think there are a couple more places where the length of hash is set to a hard coded integer instead of the macro:

* https://github.com/ARMmbed/mbedtls/pull/2208/files#diff-74f52c6a0d53e8318fbd0c437b927127R211
* https://github.com/ARMmbed/mbedtls/pull/2208/files#diff-74f52c6a0d53e8318fbd0c437b927127R150

I think the second case is OK because the test only uses SHA256. The first one also looks OK because even though it is hard coded to 64, the buffer only holds inputs from the .data file that are set to exactly that length. Because of this, I approved the PR, but  perhaps it would be good to set it to the max to avoid any future issues. Also, I wonder whether it would be good to set the hard-coded sig buffer in some tests to MBEDTLS_ECDSA_MAX_LEN where appropriate. Again, not a blocker, but a recommendation.

Reply:
@andresag01 Yes, there are a couple of places where the values are hard coded, but the hard coded values are valid hash lengths. We can change all the hard coded values in this file to defines, but this wasn't the original purpose of this PR. If you want, I'll change the values

Reply:
@RonEld: Its fine by me, thats why I approved the PR already. I just wrote down my conclusions of the code review I conducted as @Patater was asking about the situation in the rest of the file.